src/generala/utils/CombinationFinder.java

```diff
@@ -0,0 +1,240 @@
+            return;
+        }
+        combinationTreeMap.put(CombinationEnum.DOUBLE_PAIR,
+                ((biggestPair + secondBiggestPair) * PAIR_SIZE)
+                        + CombinationEnum.DOUBLE_PAIR.getScoreConst());
+
+    }
+
+    private void addFullHouse(EnumSet<CombinationEnum> playerRolledCombinations,
+                              Map<CombinationEnum, Integer> combinationTreeMap) {
+        if (hasPlayerRolledCombination(playerRolledCombinations, CombinationEnum.FULL_HOUSE)) {
+            return;
+        }
+        int pairToUseInCalculatingFullHouse;
+        if (biggestPair != biggestTriple) {
+            pairToUseInCalculatingFullHouse = biggestPair;
+        } else if (hasSecondBiggestPair()) {
+            pairToUseInCalculatingFullHouse = secondBiggestPair;
+        } else {
+            return;
+        }
+        combinationTreeMap.put(CombinationEnum.FULL_HOUSE,
+                (pairToUseInCalculatingFullHouse * PAIR_SIZE) + (biggestTriple * TRIPLE_SIZE)
+                        + CombinationEnum.FULL_HOUSE.getScoreConst());
+    }
+
+    private void addStraight(EnumSet<CombinationEnum> playerRolledCombinations,
+                             Map<CombinationEnum, Integer> combinationTreeMap) {
+        if (hasPlayerRolledCombination(playerRolledCombinations, CombinationEnum.STRAIGHT)) {
+            return;
+        }
+        combinationTreeMap.put(CombinationEnum.STRAIGHT, countStraightScore(sideForCalculatingStraightPoints));
+    }
+
+    //BOOLEAN METHODS
+    private boolean hasGenerala(int currentDieSideDuplicates) {
+        return currentDieSideDuplicates == DiceRoll.getDiceCount();
+    }
+
+    private boolean hasBiggestPair() {
+        return biggestPair != 0;
+    }
+
+    private boolean hasSecondBiggestPair() {
+        return secondBiggestPair != 0;
+    }
+
+    private boolean hasBiggestTriple() {
+        return biggestTriple != 0;
+    }
+
+    private boolean hasBiggestFourOfAKind() {
+        return biggestFourOfAKind != 0;
+    }
+
+    private boolean hasPlayerRolledCombination(EnumSet<CombinationEnum> rolledCombinations,
+                                               CombinationEnum combinationEnum) {
+        return rolledCombinations.contains(combinationEnum);
+    }
+
+    private boolean hasStraight(Set<Integer> rolledSidesSet) {
+        List<Integer> rolledSidesList = new ArrayList<>(rolledSidesSet);
+        for (int i = 0, counter = 0; i < rolledSidesList.size() - 1; i++) {
+            if (rolledSidesList.get(i) - 1 != rolledSidesList.get(i + 1)) {
+                counter = 0;
+            } else
+                counter++;
+            if (counter == STRAIGHT_SIZE - 1) {
+                sideForCalculatingStraightPoints = rolledSidesList.get(i + 1);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean canBreakEarly(int playerRolledCombinationCount, int combinationTreeMapSize) {
```

Comment:
Avoid using Java key words in method signature and variable names. In this case there must be a better name for this method.